### ISSUE TITLE
Fix warnings in setuptools 8.0

### ISSUE THREAD

**Issue by embray** [astropy/astropy-helpers#133]
As of setuptools 8.0, which introduced some new version parsing and full support for PEP 440, we get a couple warnings when running setup.py:

```
    /home/iguananaut/.virtualenvs/tmp-dea251bdca0911e7/local/lib/python2.7/site-packages/pkg_resources/__init__.py:186: RuntimeWarning: You have iterated over the result of pkg_resources.parse_version. This is a legacy behavior which is inconsistent with the new version class introduced in setuptools 8.0. That class should be used directly instead of attempting to iterate over the result.
      stacklevel=1,
    /home/iguananaut/.virtualenvs/tmp-dea251bdca0911e7/local/lib/python2.7/site-packages/setuptools/dist.py:284: UserWarning: The version specified requires normalization, consider using '1.1.dev0' instead of '1.1.dev'.
      self.metadata.version,
```

I'm not sure what the first one is aobout, I'm sure we can figure it out.  The second one is because we use "dev" version numbers like "1.1.dev".  But if I understand correctly (and I need to double check this) PEP-440 requires that dev versions have some unique specifier following it and can't just be bare.  We already generate a 'revision' number that we attach (like 1.1.dev1234) based on a count of git changesets.  But this count (unlike the githash) does not get stored in version.py for use in source dists of dev versions.  So that should be fixed as well.

**Comment by embray** [comment]
Huh. Weird.  Now I can't reproduce these...

**Comment by embray** [comment]
Oh right, at least the second one is only an issue when running out of a source tarball.

**Comment by embray** [comment]
Ah, the issue is a little more subtle than I thought.  Of course, the revcount *is* saved in version.py in the form of the _last_generated_version variable.  But it seems that when I run `./setup.py` from within an unpacked source tarball it still ends up overwriting version.py and removing that information, so that's no good.  Need to investigate why that is happening.  I think maybe for source tarballs there should be a much simplified `version.py` that only contains constants and no code to update them.  I think I kind of started taking things in that direction with c7c102898189ef5781aa5c933a848afb333110fe, but never took it all the way.

**Comment by embray** [comment]
This should also have been fixed by c0b08998e41ab0e35db7a120d2e835f19b4ad348
